presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkPhysicalResourceCalculator.java

```diff
@@ -0,0 +1,148 @@
+            log.warn(String.format("Source missing statistics, skipping automatic resource tuning."));
+            return new PhysicalResourceSettings(0, 0);
+        }
+
+        if ((inputDataInBytes > Double.MAX_VALUE) || (Double.isNaN(inputDataInBytes))) {
+            log.warn(String.format("Failed to retrieve correct size, data read=%.2f, skipping automatic resource tuning.", inputDataInBytes));
+            return new PhysicalResourceSettings(0, 0);
+        }
+
+        DataSize inputDataInGB = new DataSize(inputDataInBytes / 1000000000, DataSize.Unit.GIGABYTE);
+        QueryTypeByRunTime queryTypeByRunTime = getQueryType(getTargetedRuntimeForQuery(session));
+
+        boolean isCurrentQueryHourPlus = false;
+        if (queryTypeByRunTime.equals(QueryTypeByRunTime.NOT_DEFINED)) {
+            isCurrentQueryHourPlus = (inputDataInGB.compareTo(AVERAGE_SIZE_FOR_SHORTER_QUERIES) > 0) ? true : false;
+        }
+        else if (queryTypeByRunTime.equals(QueryTypeByRunTime.LONG)) {
+            isCurrentQueryHourPlus = true;
+        }
+
+        this.executorCount = calculateExecutorCount(session, inputDataInGB, isCurrentQueryHourPlus);
+        this.shufflePartitionCount = calculateShufflePartitionCount(session, executorCount, inputDataInGB, isCurrentQueryHourPlus);
+
+        return new PhysicalResourceSettings(executorCount, shufflePartitionCount);
+    }
+
+    private QueryTypeByRunTime getQueryType(Duration targetedRuntimeForQuery)
+    {
+        if (targetedRuntimeForQuery.compareTo(Duration.valueOf("1h")) > 0) {
+            return QueryTypeByRunTime.LONG;
+        }
+        else if (targetedRuntimeForQuery.equals(Duration.valueOf("0m"))) {
+            return QueryTypeByRunTime.NOT_DEFINED;
+        }
+        else {
+            return QueryTypeByRunTime.SMALL;
+        }
+    }
+
+    private int calculateExecutorCount(Session session, DataSize inputDataInGB, boolean isCurrentQueryHourPlus)
+    {
+        double resourceUsagePerExecutor = getResourceUsagePerExecutor(session).getValue();
+        double calculatedNumberOfExecutors = inputDataInGB.getValue() / resourceUsagePerExecutor;
+        int minExecutorCount = getMinExecutorCountForResourceAllocation(session);
+        int maxExecutorCount = getMaxExecutorForResourceAllocation(session);
+        if (false == isCurrentQueryHourPlus) {
+            minExecutorCount = (int) (minExecutorCount * SCALING_FACTOR_FOR_EXECUTOR_COUNT_BASED_ON_RUNTIME);
+            maxExecutorCount = (int) (maxExecutorCount * SCALING_FACTOR_FOR_EXECUTOR_COUNT_BASED_ON_RUNTIME);
+        }
+        int updatedExecutorCount = Math.max((int) calculatedNumberOfExecutors, minExecutorCount);
+        int tunedExecutorCount = Math.max(minExecutorCount, Math.min(maxExecutorCount, updatedExecutorCount));
+
+        return tunedExecutorCount;
+    }
+
+    private int calculateShufflePartitionCount(Session session, int executorCount, DataSize inputDataInGB, boolean isCurrentQueryHourPlus)
```

Comment:
`executorCount` and `isCurrentQueryHourPlus` are not used in this method.

Reply:
Removed